crypto/fipsmodule/sha/sha3.c

```diff
@@ -377,3 +377,48 @@ int SHAKE_Squeeze(uint8_t *md, KECCAK1600_CTX *ctx, size_t len) {
   //FIPS_service_indicator_update_state();
   return 1;
 }
+
+int SHAKE128_Init_x4(KECCAK1600_CTX_x4 *ctx) {
+
+  int ok = (SHAKE_Init(&(*ctx)[0], SHAKE128_BLOCKSIZE) &&
+            SHAKE_Init(&(*ctx)[1], SHAKE128_BLOCKSIZE) &&
+            SHAKE_Init(&(*ctx)[2], SHAKE128_BLOCKSIZE) &&
+            SHAKE_Init(&(*ctx)[3], SHAKE128_BLOCKSIZE));
+
+  return ok;
+}
+
+int SHAKE128_Absorb_once_x4(KECCAK1600_CTX_x4 *ctx, const void *data0, const void *data1,
```

Comment:
Under what condition would this function fail?

Reply:
Commented here https://github.com/aws/aws-lc/blob/4fca7ae7122f5f6d0c15625f90a3211e54f76772/crypto/fipsmodule/sha/internal.h#L473

Reply:
If the function always succeeds, can we drop the return code?

Reply:
Same as above: I think it is better to keep it. It is, indeed, a function designed specifically for mlkem, nevertheless, nothing prevents from calling it in different use case scenario (if some). Wouldn't it be better to document the _skipped_ return code check in mlkem-native, and argument it with the input values?

Reply:
Yes, but this is not a question about the caller/use case?